### PR TITLE
Fix up image sign and trust

### PR DESCRIPTION
**Pull request by rhatdan**
Add completions
Fix man pages
fix code in sign to answer PR Comments.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: completions/bash/podman [hunk 1]
```diff
@@ -2356,6 +2362,92 @@ _podman_container_runlabel() {
...
+}
+
+_podman_image_trust_show() {
+    local options_with_args="
+    "
+    local boolean_options="
+	--help
+	-h
+	-j
+        -json
```
Comment (TomSweeneyRedHat) [inline completions/bash/podman]: I think you're missing a dash here.

**Comment by TomSweeneyRedHat** [comment]
I suspect you're missing a dash in one of the completions, otherwise LGTM

**Comment by rhatdan** [comment]
@QiWang19 @mheon @baude @giuseppe @vrothberg PTAL

**Comment by TomSweeneyRedHat** [comment]
LGTM

File: cmd/podman/sign.go [hunk 2]
```diff
@@ -138,7 +138,7 @@ func signCmd(c *cli.Context) error {
 			return errors.Wrapf(err, "error creating new signature")
 		}
 
-		sigStoreDir = fmt.Sprintf("%s/%s", sigStoreDir, strings.Replace(repos[0][strings.Index(repos[0], "/")+1:len(repos[0])], ":", "=", 1))
+		sigStoreDir = filepath.Join(sigStoreDir, strings.Replace(repos[0][strings.Index(repos[0], "/")+1:len(repos[0])], ":", "=", 1))
```
Comment (vrothberg) [inline cmd/podman/sign.go]: My brain really cannot decrypt the `strings.Replace(...)` call (non-blocking).

Comment (rhatdan) [reply cmd/podman/sign.go]: @QiWang19 Please explain what this is doing.

**Comment by mheon** [comment]
This will probably need a rebase one #2122 lands - will merge once that happens

File: cmd/podman/sign.go (hunk 2 above)
Comment (QiWang19) [reply cmd/podman/sign.go]: It is just the folder name to store the signature, according to atomic it is ```/var/lib/atomic/sigstore/library/alpine@sha256=621c2f39f8133acb8e64023a94dbdf0d5ca81896102b9e57c0dc184cadaf5528```,
so I replace the ```:``` in the digest with ```=```

Comment (mheon) [reply cmd/podman/sign.go]: The strange part of this is the multi-level array indexed by a strings.Index on the first level... Is that a slice index, trying to trim repos[0]?

**Comment by rhatdan** [comment]
Looks like we survived without requiring a rebase.

File: cmd/podman/sign.go (hunk 2 above)
Comment (QiWang19) [reply cmd/podman/sign.go]: @mheon , yes...

Comment (rhatdan) [reply cmd/podman/sign.go]: @QiWang19 Break this into multi lines with variable names that tell us what is going on.  6 months from now you will have a hard time deciphering this.

**Pull request merged**